starknet_devnet/compiler.py

```diff
@@ -29,13 +29,15 @@ def compile_contract_class(self, contract_class: ContractClass) -> CompiledClass
         raise NotImplementedError
 
 
+COMPILATION_ERROR_MSG = """
+Failed compilation from Sierra to Casm! Make sure you compiled the contract with the same compiler version Devnet is using for recompilation.
+Find more info in https://0xspaceshard.github.io/starknet-devnet/docs/guide/cairo1-support/"""
```

Comment:
I'm not sure if we should have these doc links in the code.  For me, this "Find more info in URL" can be removed.

You can also merge it as it is, it's just a suggestion.